packages/web-app-files/tests/unit/components/SideBar/SideBar.spec.js

```diff
@@ -42,51 +45,327 @@ function createWrapper({ item, selectedItems, mocks }) {
+    })
   })
 
-  it('does not fetch file info if multiple items are selected', () => {
-    const mockFileInfo = jest.fn()
+  describe('when only one item is selected', () => {
+    let item, wrapper
+    afterEach(() => {
+      jest.clearAllMocks()
+    })
+
+    beforeEach(async () => {
+      item = getResource({ filename: 'testfile', extension: 'txt' })
+      wrapper = createWrapper({
+        item,
+        selectedItems: [item],
+        mocks: {
+          $client: {
+            files: {
+              fileInfo: () =>
+                getFileResponse({
+                  filename: 'testfile',
+                  extension: 'txt',
+                  sidebarPanels: samplePanels
+                })
+            }
+          }
+        }
+      })
 
-    createWrapper({
-      item: simpleOwnFolder,
-      selectedItems: [simpleOwnFolder, simpleOwnFolder],
-      mocks: { $client: { files: { fileInfo: mockFileInfo } } }
+      await flushPromises()
     })
 
-    expect(mockFileInfo).not.toHaveBeenCalled()
+    const allPanels = ['Details', 'Actions', 'People']
+
+    it.each(allPanels)('should contain all panels', expectedPanel => {
+      const panel = wrapper.find(panelSelectors[expectedPanel])
+      expect(panel.exists()).toBeTruthy()
+
+      expect(panel).toMatchSnapshot()
+    })
+
+    it('should set the active class on default panel', () => {
+      const detailPanel = wrapper.find(panelSelectors.Details)
+      expect(detailPanel.exists()).toBeTruthy()
+      expect(detailPanel.classes()).toContain('is-active-default-panel')
+    })
+
+    it('should set the active class on a panel when the panel button is clicked', async () => {
+      wrapper = createMountedWrapper({
+        item,
+        selectedItems: [item],
+        mocks: {
+          $client: {
+            files: {
+              fileInfo: () =>
+                getFileResponse({
+                  filename: 'testfile',
+                  extension: 'txt'
+                })
+            }
+          }
+        },
+        sidebarPanels: samplePanels
+      })
+
+      await flushPromises()
+
+      const buttons = {
+        Actions: selectors.actionButton,
+        People: selectors.peopleButton
+      }
+
+      for (const key in buttons) {
+        const button = wrapper.find(buttons[key])
+        expect(button.exists()).toBeTruthy()
+
+        let panel = wrapper.find(panelSelectors[key])
+        expect(panel.exists()).toBeTruthy()
+
+        const body = panel.find(selectors.panelBody)
+        expect(body.text()).toBe(key + ' content')
+        expect(panel.classes()).not.toContain('is-active-sub-panel')
+
+        await button.trigger('click')
+
+        panel = wrapper.find(panelSelectors[key])
+        expect(panel.classes()).toContain('is-active-sub-panel')
+        expect(body.text()).toContain(key + ' content')
+      }
+    })
+  })
+
+  describe('when multiple items are selected', () => {
+    it('should match the snapshot', async () => {
```

Comment:
This test lacks a purpose. Just testing if a snapshot is matched without a good reason for it makes the test suite volatile.